pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -643,27 +647,66 @@ func getLatestConfig(getconfigCtx *getconfigContext, url string,
 		return invalidConfig, rv.TracedReqs
 	}
 
+	cfgRetval := configOK
+	if !changed {
+		log.Tracef("Configuration from zedcloud is unchanged")
+		// Update modification time since checked by readSavedConfig
+		touchReceivedProtoMessage()
+		goto cfgReceived
+	}
+
+	cfgRetval = inhaleDeviceConfig(getconfigCtx, config, fromController)
+	if cfgRetval != configOK {
+		log.Errorf("inhaleDeviceConfig failed: %d", cfgRetval)
+		return cfgRetval, rv.TracedReqs
+	}
+
 	// Inform ledmanager about config received from cloud
 	utils.UpdateLedManagerConfig(log, types.LedBlinkOnboarded)
 	getconfigCtx.ledBlinkCount = types.LedBlinkOnboarded
 
-	if !getconfigCtx.configReceived {
-		getconfigCtx.configReceived = true
-	}
 	getconfigCtx.configGetStatus = types.ConfigGetSuccess
 	publishZedAgentStatus(getconfigCtx)
 
-	if !changed {
-		log.Tracef("Configuration from zedcloud is unchanged")
-		// Update modification time since checked by readSavedConfig
-		touchReceivedProtoMessage()
-		return configOK, rv.TracedReqs
-	}
-
 	// Save configuration wrapped in AuthContainer.
 	saveReceivedProtoMessage(authWrappedRV.RespContents)
 
-	return inhaleDeviceConfig(getconfigCtx, config, fromController), rv.TracedReqs
+cfgReceived:
+	getconfigCtx.configReceived = true
+
+	return configOK, rv.TracedReqs
+}
+
+// Returns true if attempt to get a configugration has failed, but initial
+// configuration was received (either from the controller, either successfully
+// read from the file)
+func needRequestLocConfig(getconfigCtx *getconfigContext,
+	rv configProcessingRetval) bool {
+
+	return (rv != configOK && getconfigCtx.locConfig != nil)
+}
```

Comment:
First I had to read this comment and function carefully to understand its semantics, then go look at where it is called/used, only to realize that it is only called in one place and if the logical expression was inline I would have saved 2 minutes in reading.

This is what I call a too shallow abstraction. I know that is very subjective.